src/crimson/os/seastore/object_data_handler.cc

```diff
@@ -263,28 +286,81 @@ overwrite_ops_t prepare_ops_list(
       assert(to_write.size() > 1);
       assert(back.addr + back.len ==
 	back.pin->get_key() + back.pin->get_length());
-      ops.to_remap.push_back(extent_to_remap_t::create_remap(
+      ops.to_remap.push_back(extent_to_remap_t::create_remap1(
 	std::move(back.pin),
 	back.addr - back.pin->get_key(),
 	back.len));
       ops.to_remove.pop_back();
     }
   }
 
-  // prepare to_insert
+  interval_set<uint64_t> pre_alloc_addr_removed, pre_alloc_addr_remapped;
+  if (delta_based_overwrite_max_extent_size) {
+    for (auto &r : ops.to_remove) {
+      if (r->is_stable() && r->get_val().is_real()) {
```

Comment:
Let's introduce a `LBAMapping::is_zero_reserved()` interface, and use it in object_data_handler wherever possible.

Reply:
Done.